projects/components/src/common/interfaces/action-item.interface.ts

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+/**
+ * List of different type of action buckets
+ */
+export enum ActionType {
+    /**
+     * Global actions that are displayed always irrespective of the context. These get shown as the first set of actions
+     */
+    STATIC_FEATURED = 'STATIC_FEATURED',
+    /**
+     * Actions that are visible when an entity is selected. These get shown up in the main menu section
+     */
+    CONTEXTUAL_FEATURED = 'CONTEXTUAL_FEATURED',
+    /**
+     * Actions that are visible when an entity is selected. These get shown up in a sub menu
+     */
+    CONTEXTUAL = 'CONTEXTUAL',
+    /**
+     * Actions that are not context dependent. These get shown up after the above 3 buckets of actions
+     */
+    STATIC = 'STATIC',
+}
+
+/**
+ * Type of {@link ActionItem.handler} function
+ */
+export type ActionHandlerType<R, T> = (selectedEntities?: R[], handlerData?: T) => Promise<string | undefined> | void;
+
+/**
+ * Data required for displaying an action item in a menu.
+ * T is the type of custom data passed to action handler
+ * R is the type of selected entity on which the action will be performed
+ */
+export interface ActionItem<R = unknown, T = unknown> {
+    /**
+     * The i18n key for the text of the button
+     */
+    textKey?: string;
+    /**
+     * The css class the button should have.
+     *
+     * @unique among all added buttons
```

Comment:
`@unique` is not a jsdoc tag, replace with

Must be unique among all added actions within an action list